#! /bin/sh /usr/share/dpatch/dpatch-run
## 24_make_dmix_dsnoop_defaults_for_ice17xx.dpatch by  <crimsun@localhost.localdomain>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make dmix & dsnoop the defaults for ICE1712 and ICE1724

@DPATCH@

diff -x '*.hg*' -uNr hg-alsa-lib-1.0.10/src/conf/cards/ICE1712.conf hg-for-pitti-alsa-lib-1.0.10/src/conf/cards/ICE1712.conf
--- hg-alsa-lib-1.0.10/src/conf/cards/ICE1712.conf	2004-01-27 04:47:13.000000000 -0500
+++ hg-for-pitti-alsa-lib-1.0.10/src/conf/cards/ICE1712.conf	2006-05-06 23:29:14.000000000 -0400
@@ -2,6 +2,29 @@
 # Configuration for the ICE1712 (Envy24) chip
 #
 
+# default with dmix & dsnoop
+ICE1712.pcm.default {
+	@args [ CARD ]
+	@args.CARD {
+		type string
+	}
+	type asym
+	playback.pcm {
+		type plug
+		slave.pcm {
+			@func concat
+			strings [ "dmix:" $CARD ",FORMAT=S32_LE" ]
+		}
+	}
+	capture.pcm {
+		type plug
+		slave.pcm {
+			@func concat
+			strings [ "dsnoop:" $CARD ",FORMAT=S32_LE" ]
+		}
+	}
+}
+
 <confdir:pcm/front.conf>
 
 ICE1712.pcm.front.0 {
diff -x '*.hg*' -uNr hg-alsa-lib-1.0.10/src/conf/cards/ICE1724.conf hg-for-pitti-alsa-lib-1.0.10/src/conf/cards/ICE1724.conf
--- hg-alsa-lib-1.0.10/src/conf/cards/ICE1724.conf	2005-09-02 12:40:40.000000000 -0400
+++ hg-for-pitti-alsa-lib-1.0.10/src/conf/cards/ICE1724.conf	2006-05-06 23:27:48.000000000 -0400
@@ -2,6 +2,29 @@
 # Configuration for the ICE1724 (Envy24HT) chip
 #
 
+# default with dmix & dsnoop
+ICE1724.pcm.default {
+	@args [ CARD ]
+	@args.CARD {
+		type string
+	}
+	type asym
+	playback.pcm {
+		type plug
+		slave.pcm {
+			@func concat
+			strings [ "dmix:" $CARD ",FORMAT=S32_LE" ]
+		}
+	}
+	capture.pcm {
+		type plug
+		slave.pcm {
+			@func concat
+			strings [ "dsnoop:" $CARD ",FORMAT=S32_LE" ]
+		}
+	}
+}
+
 <confdir:pcm/front.conf>
 
 ICE1724.pcm.front.0 {
